vocode/streaming/agent/base_agent.py

```diff
@@ -268,6 +293,42 @@ async def process(self, item: InterruptibleEvent[AgentInput]):
         except asyncio.CancelledError:
             pass
 
+    def call_function(self, function_call: FunctionCall, agent_input: AgentInput):
+        action = self.action_factory.create_action(function_call.name)
+        params = json.loads(function_call.arguments)
+        if "user_message" in params:
+            user_message = params["user_message"]
+            self.produce_interruptible_event_nonblocking(
+                AgentResponseMessage(message=BaseMessage(text=user_message))
+            )
+        action_input: ActionInput
+        if isinstance(action, VonagePhoneCallAction):
+            assert (
```

Comment:
nit but asserts should really be reserved for test cases

we should probably raise an exception instead